src/Greeting.js

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect, useState } from 'react';
+
+const Greeting = () => {
+  const [randomGreeting, setRandomGreeting] = useState('');
+
+  useEffect(() => {
+    fetch('http://127.0.0.1:3000/random_greeting')
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then((data) => setRandomGreeting(data.content))
+      .catch((error) => console.log('Error fetching greeting:', error));
+  }, []);
+
+  return (
+    <div>
+      <h1>Random Greeting:</h1>
+      <p>{randomGreeting}</p>
+    </div>
```

Comment:
- [ ] Kindly add redux to your app, and create a store, an action, and a reducer as per the project requirement.


![redux](https://github.com/Dadadon/hello-react-front-end/assets/53124548/5766216a-c996-4bff-92ee-1cccdb0cba7e)

Reply:
- As requested by the previous reviewer [here](https://github.com/Dadadon/hello-react-front-end/pull/1#discussion_r1285074178) kindly add redux to your app, and create a store, an action, and a reducer as per the project requirement

Reply:
- [ ] As has been requested, please create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting. Currently your greeting is being accessed using the `Greeting component.

Follow the following steps

1.  Create the Redux store, action, and reducer

code example

reducer
```
import {
  FETCH_GREETING_REQUEST,
  FETCH_GREETING_SUCCESS,
  FETCH_GREETING_FAILURE,
} from './actions';

const initialState = {
  greeting: '',
  loading: false,
  error: null,
};

const greetingReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_GREETING_REQUEST:
      return {
        ...state,
        loading: true,
        error: null,
      };
    case FETCH_GREETING_SUCCESS:
      return {
        ...state,
        loading: false,
        greeting: action.payload,
      };
    case FETCH_GREETING_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };
    default:
      return state;
  }
};

export default greetingReducer;
```

actions
```
import axios from 'axios';

export const FETCH_GREETING_REQUEST = 'FETCH_GREETING_REQUEST';
export const FETCH_GREETING_SUCCESS = 'FETCH_GREETING_SUCCESS';
export const FETCH_GREETING_FAILURE = 'FETCH_GREETING_FAILURE';

const apiEndpoint = 'https://your-rails-api.com/api/random-greeting';

const fetchGreetingRequest = () => ({
  type: FETCH_GREETING_REQUEST,
});

const fetchGreetingSuccess = (greeting) => ({
  type: FETCH_GREETING_SUCCESS,
  payload: greeting,
});

const fetchGreetingFailure = (error) => ({
  type: FETCH_GREETING_FAILURE,
  payload: error,
});

export const fetchRandomGreeting = () => (dispatch) => {
  dispatch(fetchGreetingRequest());
  axios
    .get(apiEndpoint)
    .then((response) => {
      const randomGreeting = response.data.greeting;
      dispatch(fetchGreetingSuccess(randomGreeting));
    })
    .catch((error) => {
      dispatch(fetchGreetingFailure(error.message));
    });
};
```


store
```
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
```

2. Connect components to the Redux store using the `Provider` component from react-redux:

example

```
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import App from './App';
import store from './store';

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
);
```

3. Finally you can use the component to display the random greeting:

example

```
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { fetchRandomGreeting } from './actions';

const Greeting = ({ greeting, loading, error, fetchRandomGreeting }) => {
  useEffect(() => {
    fetchRandomGreeting();
  }, [fetchRandomGreeting]);

  if (loading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return <p>Error: {error}</p>;
  }

  return <p>{greeting}</p>;
};

const mapStateToProps = (state) => ({
  greeting: state.greeting,
  loading: state.loading,
  error: state.error,
});

const mapDispatchToProps = {
  fetchRandomGreeting,
};

export default connect(mapStateToProps, mapDispatchToProps)(Greeting);
```



### Note that this is an example